client/src/components/carousel.jsx

```diff
@@ -32,17 +38,72 @@ class Carousel extends React.Component {
     });
   }
 
+  toggleModal() {
+    const { displayModal: display } = this.state;
+    this.setState({
+      displayModal: !display,
+    });
+  }
+
+  handleClickedImage(clickedIndex, callback) {
+    this.setState({
+      clickedIdx: clickedIndex,
+    }, callback);
+  }
+
+  handleModalNextButton() {
+    let { clickedIdx: idx } = this.state;
+    const { data: dataArr } = this.state;
+    if (String(idx)
+    === String(dataArr.photosAndComments.length - 1)) {
+      this.setState({
+        clickedIdx: 0,
+      });
+    } else {
+      this.setState({
+        clickedIdx: idx += 1,
+      });
+    }
+  }
+
+  handleModalPreviousButton() {
+    let { clickedIdx: idx } = this.state;
+    const { data: dataArr } = this.state;
+    if (String(idx) === String(0)) {
+      this.setState({
+        clickedIdx: dataArr.photosAndComments.length - 1,
+      });
+    } else {
+      this.setState({
+        clickedIdx: idx -= 1,
+      });
+    }
+  }
+
   render() {
     const { data: dataArr } = this.state;
+    const { displayModal: display } = this.state;
+    const { clickedIdx: idx } = this.state;
     if (!dataArr) {
```

Comment:
Lines 84, 85, 86  can be condensed into one variable like this `const { data, displayModal, clickedIdx } = this.state` It's not necessary to list the property of the key in the destructure.